ts/src/vertex.ts

```diff
@@ -0,0 +1,1741 @@
+        }
+        return status;
+    }
+
+    parseOrder (order, market: Market = undefined): Order {
+        //
+        // {
+        //     "product_id": 1,
+        //     "sender": "0x7a5ec2748e9065794491a8d29dcf3f9edb8d7c43000000000000000000000000",
+        //     "price_x18": "1000000000000000000",
+        //     "amount": "1000000000000000000",
+        //     "expiration": "2000000000",
+        //     "nonce": "1",
+        //     "unfilled_amount": "1000000000000000000",
+        //     "digest": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        //     "placed_at": 1681951347,
+        //     "order_type": "ioc"
+        // }
+        // stop order
+        // {
+        //     "order": {
+        //       "order": {
+        //         "sender": "0x7a5ec2748e9065794491a8d29dcf3f9edb8d7c43000000000000000000000000",
+        //         "priceX18": "1000000000000000000",
+        //         "amount": "1000000000000000000",
+        //         "expiration": "2000000000",
+        //         "nonce": "1",
+        //       },
+        //       "signature": "0x...",
+        //       "product_id": 1,
+        //       "spot_leverage": true,
+        //       "trigger": {
+        //         "price_above": "1000000000000000000"
+        //       },
+        //       "digest": "0x..."
+        //     },
+        //     "status": "pending",
+        //     "updated_at": 1688768157050
+        // }
+        //
+        let marketId = this.safeString (order, 'product_id');
+        let timestamp = this.safeTimestamp (order, 'placed_at');
+        let amount = this.safeString (order, 'amount');
+        let price = this.safeString (order, 'price_X18');
+        let remaining = this.safeString (order, 'unfilled_amount');
+        let triggerPriceNum = undefined;
+        const status = this.safeString (order, 'status');
+        if (status !== undefined) {
+            // trigger order
+            const outerOrder = this.safeDict (order, 'order', {});
+            const innerOrder = this.safeDict (outerOrder, 'order', {});
+            marketId = this.safeString (outerOrder, 'product_id');
+            amount = this.safeString (innerOrder, 'amount');
+            price = this.safeString (innerOrder, 'priceX18');
+            timestamp = this.safeTimestamp(order, 'updated_at');
+            const trigger = this.safeDict (outerOrder, 'trigger', {});
+            const triggerPrice = this.safeStringN (trigger, [ 'price_above', 'price_below', 'last_price_above', 'last_price_below' ]);
+            if (triggerPrice !== undefined) {
+                triggerPriceNum = this.parseToNumeric (this.convertFromX18 (triggerPrice));
+            }
+        }
+        market = this.safeMarket (marketId, market);
+        const symbol = market['symbol'];
+        let priceNum = undefined;
+        if (price !== undefined) {
+            priceNum = this.parseToNumeric (this.convertFromX18 (price));
```

Comment:
we don't need to convert amount/price/etc to numeric, they can remain as strings and safeOrder will parse them accordingly.